desktop/server/server.js

```diff
@@ -25,18 +25,47 @@ function showFailure( app ) {
 	} );
 }
 
-function startServer() {
-	var boot = require( 'boot' );
-	var http = require( 'http' );
-	var server = http.createServer( boot() );
+function startServer( startedCallback ) {
+	var path = require( 'path' ),
+		fork = require( 'child_process' ).fork,
+		appFile = path.resolve( process.cwd(), 'calypso', 'build', 'bundle-desktop.js' ),
+		env = Object.create( process.env ),
+		startedSignal = 'wp-calypso booted',
+		app;
 
-	debug( 'Server created, binding to ' + Config.server_port );
+	env.PORT = Config.server_port;
+	env.NODE_PATH = 'server:client:.';
 
-	server.listen( Config.server_port, Config.server_host );
+	debug( 'Loading server from file: ' + appFile );
+
+	app = fork( appFile, [], {
+		silent: true,
+		cwd: path.resolve( process.cwd(), 'calypso' ), // Force working directory to calypso root
+		env: env
+	} );
+
+	app.stdout.on( 'data', function( data ) {
+		// We need to wait until the server has booted.
+		// Let's use the "booted" message as a signal that it's ready.
+		var str = data.toString();
+		if ( str.indexOf( startedSignal ) === 0 ) {
```

Comment:
This is a bit annoying and fragile. If Calypso ever changes the wording of the boot message, the UI will fail to load.